sql_upgrade.php

```diff
@@ -70,52 +72,141 @@
+        // exceeds 98% or end of upgrade script. New script restarts.
+        async function serverStatus() {
+            let url = "library/ajax/sql_server_status.php?poll=";
+            url += encodeURIComponent(doPoll);
+            let response = await fetch(url);
+
+            if (response.status === 502) {
+                // connection timeout, just reconnect
+                if (doPoll) {
+                    await serverStatus();
+                }
+            } else if (response.status !== 200) {
+                // show error
+                progressStatus(response.statusText);
+                // reconnect in one second
+                await new Promise(resolve => setTimeout(resolve, 1000));
+                if (doPoll) {
+                    await serverStatus();
+                }
+            } else {
+                // await status
+                let status = await response.text();
+                // display to screen div
+                progressStatus(status);
+                // and so forth.
+                if (doPoll) {
+                    await serverStatus();
+                } else {
+                    progressStatus("<li>Done this upgrade</li>");
+                }
+            }
+        }
+
+
+        function progressStatus(msg) {
+            if (processProgress > 98) {
+                doPoll = 0;
+                processProgress = 100;
+            }
+            let eventList = document.getElementById('status-message');
+            let progressEl = document.getElementById('progress');
+            progressEl.style.width = processProgress + "%";
+            progressEl.innerHTML = processProgress + "%";
+            eventList.innerHTML += msg;
+        }
+
+        function setWarnings(othis) {
+            if (othis.value < '5.0.0') {
+                document.querySelector('.version-warning').classList.remove("d-none");
+            } else {
+                document.querySelector('.version-warning').classList.add("d-none");
+            }
+        }
+
+        window.onload = (event) => {
+            // I'll eventually find a reason to use this!
+        }
+    </script>
 </head>
 <body>
-    <div class="container mt-3">
+    <div class="container my-3">
         <div class="row">
             <div class="col-12">
-                <h2>OpenEMR Database Upgrade</h2>
+                <h2><?php echo xlt("OpenEMR Database Upgrade"); ?></h2>
             </div>
         </div>
-        <div class="jumbotron p-4">
+        <div>
             <form class="form-inline" method='post' action='sql_upgrade.php'>
-                <div class="form-group mb-3">
-                    <label>Please select the prior release you are converting from:</label>
-                    <select class='ml-3 form-control' name='form_old_version'>
+                <div class="form-group mb-1">
+                    <label><?php echo xlt("Please select the prior release you are converting from"); ?>:</label>
```

Comment:
One thing to always keep in mind is this script is running a "special" OpenEMR where the codebase is current but the database foorprint can be really old. This is the reason never added translations since wasn't sure where it may break in prior openemr database footprints (ie. if the query in the current codebase may have columns not in the old database footprint). However, the translation query and database elements have been stable for a long time (the only thing i thin added over last 10 years has been the rtl flag, which is not even used in the query, so I think we are at a point where running translations in this script is safe and makes sense. Thanks for doing this.

Reply:
Good point Brady. During doing this it was always in the back of my mind of hum, i'd of thought Brady would have done this by now so, now I know why. thanks.